### DISCUSSION TITLE
Can I avoid label printing?

### DISCUSSION THREAD

**Discussion by grammenoudis** [inventree/InvenTree#8125] [Q&A]
Can I somehow reuse the mouser 2d barcodes instead of printing and scanning new barcodes in order to manage stock?
If there is a large amount of components am I expected to print x labels and stick them to the assigned containers?

My thought was to reuse the 2d barcode to retrieve the part number and query the results based on that instead of the data given by the generated labels ( example {"part" : 51} )

example output of 2d barcode:
<img width="1576" alt="Screenshot 2024-09-13 at 7 35 16 PM" src="https://github.com/user-attachments/assets/f5d9dce1-fef1-4811-9411-acb710a413c2">

**Comment by SchrodingersGat** [comment]
Yes you can assign a third party barcode to an existing stock item. 

https://docs.inventree.org/en/latest/barcodes/external/

You can do this via the app or the web interface

**Reply by grammenoudis** [reply]
That would still involve scanning every barcode to link the items, that sounds time consuming if you have many. And if I have many mouser packets with the same item, that would require the same items to have multiple linked barcodes since I want to have access from all.

I was thinking of writing a plugin that checks if the 2d barcode has ever been seen before, if not, add the product to stock otherwise use the part number to navigate to the correct web page.

Is this possible? I'm not familiar with the source code so I have a hard time with it. This is the only change I need to make for our team to use Inventree the way we want

**Reply by grammenoudis** [reply]
By taking a good look at the demo I see that the feature that I want for detecting the parts with the original barcode is built in if I use the data extracted by the barcode when creating the part via the web interface (example data: 'SKU [CAP_01A]:L1,L2', supplier 'DigiKey', MPN 'CMH322522-180KL')

Now what I need is the ability to automatically add stock if the barcode is read for the first time, is this a built in feature that i can integrate with the other described? (a receive order feature like the mouser inventory website would solve my problems. Can this be taken care of with the mobile app or the website?)

**Reply by SchrodingersGat** [reply]
Yes you can "check stock in" against manufacturer barcodes (i.e. digikey / mouser / etc)

**Reply by grammenoudis** [reply]
I can't find a "check stock in" option somewhere in the webapp, could you point me to it if you don't mind?

**Reply by SchrodingersGat** [reply]
@grammenoudis only via the mobile app currently. We will be integrating this feature into the new [React interface](https://github.com/inventree/InvenTree/issues/5212) but constrained by available developer time currently

**Reply by grammenoudis** [reply]
@SchrodingersGat now that i downloaded the apk, when i go Purchase Orders > Scan Received Parts it says "No match found for supplier barcode" when I scan a recent mouser 2d barcode, do I have to first add the order manually for it to "receive" it or  am I doing something wrong?

**Reply by SchrodingersGat** [reply]
@grammenoudis I am working on a solution for debugging barcode scanning. Please see this PR: https://github.com/inventree/InvenTree/pull/8150
